code/game/objects/map_metadata/save_awards.dm

```diff
@@ -15,52 +15,60 @@ AWARDS:
-					map.scores["Western Army"] += 1
-					awards["service"]=0
-	
-	if (map.ID == MAP_BANK_ROBBERY || map.ID == MAP_DRUG_BUST)
-		if (awards["arrests"]>=3 && client && !("bronze public order ribbon" in awarded))
-			map.give_award(client.ckey, name, "bronze public order ribbon", capitalize(faction_text),src)
-			return TRUE
-
-	if (map.gamemode == "Hardcore")
-		if (map.ID == MAP_BANK_ROBBERY || map.ID == MAP_DRUG_BUST) //Prevents farming kill medals on those maps
-			return FALSE
-		else
-			if (map.ordinal_age>=5)
-				awards["service"]++
-				if (awards["service"]>=1200 && client && !("long service medal" in awarded))
-					map.give_award(client.ckey, name, "long service medal", capitalize(faction_text),src)
-
-				if (awards["medic"]>=200 && client && !("medical medal" in awarded))
-					map.give_award(client.ckey, name, "medical medal", capitalize(faction_text),src)
-
-				if (awards["tank"]>=2 && !("tank destroyer silver badge" in awarded))
-					map.give_award(client.ckey, name, "tank destroyer silver badge", capitalize(faction_text),src)
-				else if (awards["tank"]>=4 && !("tank destroyer gold badge" in awarded))
-					map.give_award(client.ckey, name, "tank destroyer gold badge", capitalize(faction_text),src)
-					map.remove_award(client.ckey, name, "tank destroyer silver badge")
+    if (!client) return FALSE
+
+    switch(map.ID)
+        if (MAP_ALLEYWAY)
+            if (stat != DEAD)
+                if (original_job && (original_job.title == "Yama Wakagashira" || original_job.title == "Ichi Wakagashira"))
+                    awards["service"]++
+                    if (awards["service"] >= 60)
+                        if (original_job.title == "Yama Wakagashira")
+                            map.scores["Yamaguchi-Gumi"] += 1
+                            awards["service"] = 0
+                        else if (original_job.title == "Ichi Wakagashira")
+                            map.scores["Ichiwa-Kai"] += 1
+                            awards["service"] = 0
+        
+        if (MAP_SEKIGAHARA)
+            if (stat != DEAD)
+                if (original_job && (original_job.title == "Azuma no Daimyo" || original_job.title == "Sei no Daimyo"))
+                    awards["service"]++
+                    if (awards["service"] >= 60)
+                        if (original_job.title == "Azuma no Daimyo")
+                            map.scores["Eastern Army"] += 1
+                            awards["service"] = 0
+                        else if (original_job.title == "Sei no Daimyo")
+                            map.scores["Western Army"] += 1
+                            awards["service"] = 0
+        
+        if (MAP_BANK_ROBBERY || MAP_DRUG_BUST)
+            if (awards["arrests"] >= 3 && client && !("bronze public order ribbon" in awarded))
+                map.give_award(client.ckey, name, "bronze public order ribbon", capitalize(faction_text), src)
+                return TRUE
+
```

Comment:
Actually might need an "else" statement here, tabulate one indentation to the right for lines 52 to 107 and remove lines 53-55. This way the 4 maps above don't get the other medal types, whatever the gamemode is.

This *might* also explain why BANK_ROBBERY and DRUG_BUST didn't get their specific medals on Hardcore. Unless awards["arrests"] remained to the value of 0, then the problem would be in the metadata of both maps.